lib/elixir/test/elixir/string_test.exs

```diff
@@ -274,6 +274,8 @@ defmodule StringTest do
 
     assert String.replace("a,b,c", ~r/,(.)/, ",\\1\\1") == "a,bb,cc"
     assert String.replace("a,b,c", ~r/,(.)/, ",\\1\\1", global: false) == "a,bb,c"
+    assert String.replace("a.b.c", ~r/\./, %{"." => "-"}) == "a-b-c"
+    assert String.replace("abba", ~r/a|b/, %{"a" => "1", "b" => "2"}) == "1221"
```

Comment:
Thank for the quick response Jose!

I quickly responded earlier that it does work when the second argument is a string and did not see what you meant :)

Thank you again for the feedback!